extensions/amp-lightbox-gallery-detection/0.1/amp-lightbox-gallery-detection.js

```diff
@@ -0,0 +1,290 @@
+  const renderArea = renderWidth * renderHeight;
+
+  const isShrunk =
+    (naturalArea / renderArea) >= RENDER_AREA_RATIO;
+
+  const isCoveringSignificantArea =
+    (renderArea / viewportArea) >= VIEWPORT_AREA_RATIO;
+
+  const isLargerThanViewport = naturalWidth > vw || naturalHeight > vh;
+
+  return isShrunk || isLargerThanViewport || isCoveringSignificantArea;
+}
+
+
+/** @visibleForTesting */
+export class Scanner {
+
+  /**
+   * @param {!Document} doc
+   * @return {!Promise<!Array<!Element>>}
+   */
+  static getAllImages(doc) {
+    const imgs = toArray(doc.querySelectorAll('amp-img'))
+        .filter(img => !img.hasAttribute(LIGHTBOXABLE_ATTR));
+
+    const promises = imgs.map(whenLoadedOrNull);
+
+    return Promise.all(promises).then(imagesOrNull =>
+      imagesOrNull.filter(i => !!i));
+  }
+}
+
+
+/**
+ * @param {!Element} element
+ * @return {boolean}
+ * @visibleForTesting
+ */
+export function meetsCriteria(element) {
+  return Criteria.meetsAll(element);
+}
+
+
+/**
+ * @param {!IArrayLike<T>} haystack
+ * @param {function(T):boolean} needleCb
+ * @return {?T}
+ * @template T
+ */
+function find(haystack, needleCb) {
+  for (let i = 0; i < haystack.length; i++) {
+    if (needleCb(haystack[i])) {
+      return haystack[i];
+    }
+  }
+  return null;
+}
+
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {boolean}
+ * @visibleForTesting
+ */
+export function isEnabledForDoc(ampdoc) {
+  if (!Services.viewerForDoc(ampdoc).isEmbedded()) {
+    return false;
+  }
+
+  const scriptTags = ampdoc.getHeadNode().querySelectorAll('script');
+
+  const schemaTag = find(scriptTags,
+      t => t.getAttribute('type') == 'application/ld+json');
+
+  const currentScriptTag = find(scriptTags,
+      t => t.getAttribute('custom-element') == REQUIRED_EXTENSION);
+
+  const lightboxedElementsSelector = `[${LIGHTBOXABLE_ATTR}]`;
+
+  if (currentScriptTag &&
+      ampdoc.getRootNode().querySelector(lightboxedElementsSelector)) {
+    return false;
+  }
+
+  if (!schemaTag) {
+    return false;
+  }
+
+  const parsed = tryParseJson(schemaTag./*OK*/innerText);
+
+  if (!parsed) {
+    return false;
+  }
+
+  return [
+    'Article',
+    'NewsArticle',
+    'BlogPosting',
+    'LiveBlogPosting',
+    'DiscussionForumPosting',
+  ].includes(parsed['@type']);
+}
+
+
+/**
+ * @param {!Element} el
+ * @return {!Promise<?Element>}
+ */
+function whenLoadedOrNull(el) {
+  return new Promise(resolve => {
+    el.signals().whenSignal(CommonSignals.LOAD_END)
+        .then(() => { resolve(el); })
+        .catch(() => { resolve(null); });
+  });
+}
+
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Array<!Element>} images
+ */
+function applyToScanned(ampdoc, images) {
+  const lightboxable = images.filter(meetsCriteria);
+
+  if (lightboxable.length <= 0) {
+    return;
+  }
+
+  iterateCursor(lightboxable, element => {
+    element.setAttribute(LIGHTBOXABLE_ATTR, '');
```

Comment:
you wanna generate a unique "group" key here as set it as the value of `lightbox=<group>` otherwise they all end up in the "default" group.